### DISCUSSION TITLE
How to calculate the volume fraction of twins from pole figure data in MTEX?

### DISCUSSION THREAD

**Discussion by MWalag** [mtex-toolbox/mtex#1352] [Ask Anything]
I was told that it is possible to calculate the volume fraction of twins from the pole figure data. I'm struggling with fixing an orientation to 'volume' function in MTEX. I have (0002) pole figures with noticeable rotation of basal planes towards compression direction which are associated with tensile twinning in hexagonal structure (Mg in my case, see attached pdf, pole figure nr. 3). My question is: volume of which orientation (I prefer to operate in rotation by.AxisAngle) should I calculate to obtain a volume fraction of twins in deformed material? At this moment I don't know that I'm wrong with MTEX code or interpretation of stereographic projection.
Example code:
```
%uploading pole figure data
odf=calcODF(pf)
rot=rotation.byAxisAngle(yvector,70*degree)
ori=orientation(rot,CS)
volume(odf,ori,3)
```
The angles which i tried to calculate volume were 40° (angle between 0002 and 10-12) and 86° (rotation of basal planes associated with twinning). In both cases results seems to be wrong, for example ~30% of twins in undeformed material in which twins are not seen in microstructure.
[polefiguresmg.pdf](https://github.com/mtex-toolbox/mtex/files/8094784/polefiguresmg.pdf)

**Comment by nyyssont** [comment]
Could your angular tolerance in the volume command be too generous? Perhaps you would like it to be 3*degree instead of 3 in radians.

**Reply by MWalag** [reply]
Of course the toleration should be in degrees instead of radians! I always forget that default unit in matlab is radian. Thank you!
